ruby/ql/test/library-tests/dataflow/array-flow/array_flow.rb

```diff
@@ -0,0 +1,523 @@
+def m0(i)
+    a = *source(0)
+    sink(a[0]) # $ hasValueFlow=0
+    sink(a[1])
+    sink(a[i]) # $ hasValueFlow=0
+end
+
+def m1(i)
+    a = [0, source(1), 2]
+    sink(a[0])
+    sink(a[1]) # $ hasValueFlow=1
+    sink(a[2])
+    sink(a[i]) # $ hasValueFlow=1
+end
+
+def m2(i)
+    a = Array.new(0, source(2.1))
```

Comment:
Dunno if this was intentional, but `Array.new(0)` will create an array of size zero - i.e. `[]`. So in reality this array won't contain the source value.